### DISCUSSION TITLE
Please help me understand how Budibase works?

### DISCUSSION THREAD

**Discussion by sangdth** [Budibase/budibase#7828] [Help] [answered]
So after working hard on our app, I face a big issue. The app does not reflect the current data in Budibase DB.

I have an automation workflow that auto insert/update the Budibase DB, and as you can see in the screenshot, the Budibase DB can have extra new row, but the app still show only 2. It only show if I click the Publish button manually again. This is so wrong in my opinion. I have tried to clear cache but still does not help.

<img width="634" alt="Screenshot 2022-09-18 at 13 07 44" src="https://user-images.githubusercontent.com/1083478/190896973-4cdb4280-98e1-4e6b-8de6-a04441f1f40d.png">

<img width="431" alt="Screenshot 2022-09-18 at 13 08 24" src="https://user-images.githubusercontent.com/1083478/190896986-5c4ff3f5-6d1f-41db-8b93-8ae29ac2d269.png">

Please help me make the app shows correct data from DB.

Thank you. 🙏🏻

**Comment by shogunpurple** [comment, marked as answer]
Hi @sangdth,

Sorry to hear you are having issues. Would you be able to show what your automation looks like? 

Are you creating the rows from within the data UI of the builder? It would be good to know the exact steps you have taken. 

Unless using the test mode provided in the automations tab, Automations only run in production apps. If you have an automation that triggers when a row is created, that automation only works when you create a row in your published, live application. Anything you create inside the builder is only stored in the development database.

The data sync between the production and development databases for an app is one way - the dev DB (the one you access from the data UI) will sync any new data you add to the production app, but if you make changes in the dev DB, you can only update the prod DB by clicking "Publish". This makes sense - you wouldn't want to automatically edit data and change the schema of a production database.

**Reply by sangdth** [reply]
Thanks @shogunpurple.

But I'm not changing the schema at this moment, I have problem with rows data.

Currently I have an n8n workflow to make POST/PUT requests to  `https://budibase.app/api/public/v1/tables.../rows`, and it adds/updates the rows correctly in Budibase DB.

I have just tried to manually use `Create row` to add a new row, but the app also can not see it too.

**Reply by sangdth** [reply]
OK sorry for misunderstanding. Now I understanding the dev/prod idea of Budibase. I created an automation inside Budibase and after publishing it works as expected.

Thank you.
